sopel/modules/admin.py

```diff
@@ -327,12 +327,22 @@ def set_config(bot, trigger):
         bot.say(exc.args[1])
         return
 
+    # Get a descriptor class for the option if it's a static section
+    descriptor = getattr(section.__class__, option) if static_sec else None
+
     # Display current value if no value is given
     if not value:
-        if option.endswith("password") or option.endswith("pass"):
+        value = getattr(section, option)
+
+        if descriptor is not None:
+            if getattr(descriptor, 'is_secret', False):
+                # Keep secret option as secret
+                value = "(password censored)"
+        elif option.endswith("password") or option.endswith("pass"):
+            # Fallback to guessing if secret, for backward compatiblity
+            # TODO: consider a deprecation warning at some point
```

Comment:
I like the idea of warning about things that look like passwords, but maybe it shouldn't be _here_.

What do you think of emitting it during settings load?

Reply:
Oh yes, that's a good idea.

Reply:
```suggestion
            # TODO: consider a deprecation warning when loading settings
```